### ISSUE TITLE
[BUG] scoped registry is not respected

### ISSUE THREAD

**Issue by pgsandstrom** [npm/npm-registry-fetch#24]
# What / Why
When adding a scoped registry to my configuration, it does not appear to be respected.

## When
```
  const conf = {
    cache: null,
    '@babel:registry': 'http://this-should-fail.example.com',
  }
  const json = (await npmRegistryFetch.json(
    '@babel/plugin-proposal-class-properties',
    conf,
  ))
```

This request succeeds when I would expect it to fail.

I have tried both with v6.0.2 and v7.0.0 and get the same result.

**Comment by pgsandstrom** [comment]
After digging around in the source code I now understand this is intended behaviour. I just failed to parse the documentation correctly.

However, I am a bit confused why this is the intended behaviour. I am currently using [libnpmconfig](https://github.com/npm/libnpmconfig) to parse my npm config, and it would be very nice to simply be able to pass this object to npm-registry-fetch. But to replicate the behaviour of the npm client it seems as if I would have to do something like this:

```
const correctConf = {...conf, spec: dependencyName}
const json = (await npmRegistryFetch.json(dependencyName, correctConf))
```

Wouldn't it be nicer to skip specifying spec? It feels as if my use case would be quite common.

**Comment by isaacs** [comment]
Yeah, it would be nice for it to parse the path and see that it's a scoped package name, and do the thing you're expecting here.

The hazard I could see is that it might actually be saving us from some sticky name collisions as it stands right now.  I'd want to make sure we're not breaking anything that folks might be depending on (even unknowingly depending on by accident).
